doc/README-mosquitto.md

```diff
@@ -0,0 +1,151 @@
+# Ecplipse mosquitto_password cracking
+
+mosquitto2john.py is a helper script to allow you to convert Eclipse Mosquitto
+passwd files into a John friendly format. The script takes a 
+[mosquitto_passwd file](https://mosquitto.org/man/mosquitto_passwd-1.html) as 
+its only required input. 
+
+## Usage
+
+```
+❯ mosquitto2john.py --help
+usage: mosquitto2john.py [-h] [-hc] [-o OUT_FILE] passwd_file
+
+positional arguments:
+  passwd_file           Path to the target mosquitto_passwd file.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -hc, --hashcat        Convert hashes to hashcat friendly formats.
+  -o OUT_FILE, --out-file OUT_FILE
+                        File name for converted hashes - required only for 
+			mixed hashes.
+
+```
+
+Whilst options are available for hashcat conversion and out_file writing, the
+script is intended to work in the classic something2john.py way, piping Stdout 
+straight into an outfile of your choosing:
+
+```
+❯ mosquitto2john.py mosquitto_passwd > mosquitto.hash
+❯ john ./mosquitto.hash
+[...SNIP...]
+Session completed. 
+```
+### Hashcat
+
+The hashcat option converts the hashes to Hashcat friendly formats - mode 1710 
+for SHA512 and 12100 for pbkdf2-hmac-sha512 hashes. In the following example, 
+some SHA512 and pbkdf2-hmac-sha512 hashes have already been exported into 
+separate files according to their type:
+
+```
+❯ ls
+sha512_hashes     HMAC_hashes
+# Identify the SHA512 hashes by their format: HEX_DIGEST:HEX_SALT
+❯ head -n 1 sha512_hashes
+SOMELONGHEXDIGEST:SOMEHEXSALT
+❯ hashcat -a 0 -m 1710 --hex-salt ./sha512_hashes [WORDLIST]
+[... SNIP ...]
+# Identify the HMAC hashes by a leading sha512: and following iteration count
+# These also have base64 salt and digests
+❯ head -n 1 HMAC_hashes
+sha512:101:BaSe64SaLt:BaSe64DiGeSt==
+❯ hashcat -a 0 -m 12100 ./HMAC_hashes [WORDLIST]
+[... SNIP ...]
+```
+
+**Note** the *use of --hex-salt in mode 1710* this is required for successful
+cracking.
+
+## Troubleshooting
+
+### No hashes loaded?
+
+If your individual install isn't automatically detecting the hash type, you 
+can try manually checking the hash format and specifying it with *--format*
+on the John command line:
```

Comment:
This should only be needed for "certain older" (you can use this wording) versions of JtR and only for "dynamic_82". You don't need to provide an example for "pbkdf2-hmac-sha512" because those are always auto-detected. So you can simplify this section omitting the instructions for how to extract the hash type and suggesting specifically the use of `--format=dynamic_82` as a workaround.